src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -160,44 +180,80 @@ public JavaSurfaceTransformer(
   }
 
   private StaticLangFileView<StaticLangApiView> generateApiFile(
-      InterfaceContext context, GapicProductConfig productConfig) {
-    StaticLangFileView.Builder<StaticLangApiView> apiFile =
-        StaticLangFileView.<StaticLangApiView>newBuilder();
+      InterfaceContext context,
+      ApiAndSamples<StaticLangApiView, StaticLangApiMethodView> classView) {
```

Comment:
This feels weird to me - it the scope of the input is too large for the method. It should either be pure source input (the original structure, where it uses the API model from the context) or just the class view, not class view + sample view.

Reply:
This is in part in response to your concern about my first approach, which generating the client surface and then reading that to generate the samples. Let's discuss.

Reply:
ptal